src/openrct2/actions/TrackDesignAction.cpp

```diff
@@ -259,7 +259,7 @@ GameActions::Result TrackDesignAction::Execute() const
 
     for (size_t i = 0; i <= MAX_VEHICLES_PER_RIDE; i++)
     {
-        auto tdIndex = std::min(i, std::size(_td.vehicle_colours) - 1);
+        auto tdIndex = i % (std::size(_td.vehicle_colours) - 1);
```

Comment:
This is really just a limitation of the td6 format but you need to consider the 3 different modes of colouring. All the same colour, different per train, different per car. You should only wrap on the last two.

Reply:
```suggestion
        uint32_t tdIndex;
        switch (ride->colour_scheme_type)
        {
            case RIDE_COLOUR_SCHEME_DIFFERENT_PER_CAR:
            case RIDE_COLOUR_SCHEME_DIFFERENT_PER_TRAIN:
                tdIndex = i % (std::size(_td.vehicle_colours) - 1);
                break;
            case RIDE_COLOUR_SCHEME_ALL_SAME:
                tdIndex = std::min(i, std::size(_td.vehicle_colours) - 1);
                break;
        }
```
Do you mean something like that?

Reply:
hmm tbh i don't think it matters too much so lets just keep it like this.